base/string.jl

```diff
@@ -2,7 +2,11 @@
 
 print(io::IO, x) = show(io, x)
 print(io::IO, xs...) = for x in xs print(io, x) end
```

Comment:
Just realized that a lock only on ` print(xs...)` will handle the `println` case too. Have updated the PR. Thanks.